tests/pass/permissive_provenance_sb.rs

```diff
@@ -0,0 +1,62 @@
+// compile-flags: -Zmiri-permissive-provenance
+
+
+// random agglomeration test
+use core::cell::UnsafeCell;
+
+fn not_so_innocent(x: &mut UnsafeCell<u32>) {
+    let x_raw: *mut UnsafeCell<u32> = x as *mut UnsafeCell<u32>;
+    x_raw as usize;
+}
+
+fn main() {
+    unsafe { 
+        let mut c = UnsafeCell::new(42u32);
+
+        let x_unique1: &mut UnsafeCell<u32> = &mut c;
+        // [..., Unique(1)]
+        
+        let x_raw1: *mut UnsafeCell<u32> = x_unique1 as *mut UnsafeCell<u32>;
+        let x_raw1_usize = x_raw1 as usize;
+        // [..., Unique(1), SharedRW(2*)]
+        // * denotes exposed
+        
+        let x_unique2: &mut UnsafeCell<u32> = &mut *x_raw1;
+        // [.., Unique(1), SharedRW(2*), Unique(3)]
+        
+        not_so_innocent(x_unique2);
+        // [.., Unique(1), SharedRW(2*), Unique(3), ..., SharedRW(4*)]
+        // * denotes exposed
+    
+        // There will be an additional SharedRW(4*) on top of Unique(3) at this
+        // point, if `not_so_innocent` is called.
+    
+        // Cast the integer to the pointer, where confusion may occur about which tag we should use.
+        let x_sharedrw = x_raw1_usize as *mut u32;
+        let x_sharedrw = &*(x_sharedrw as *const UnsafeCell<u32>);
+        let x_sharedrw = &*(x_sharedrw as *const UnsafeCell<u32>);
+    
+        // Now try to write to the casted SharedRW. This will use the tag SharedRW(4*)
+        // if not_so_innocent was called, else it will use SharedRW(2*).
+        {
+            *x_sharedrw.get() = 12;
+        }
+        
+        // Now write to the topmost Unique. If not_so_innocent was not called,
+        // this will fail.
+        {
+            *x_unique2 = UnsafeCell::new(10);
+        }
+
+        // Wildcard tags are not magic; popping all exposed tags prevents them from accessing.
+        // Uncomment this to see the next write fail.
+        // {
+        //     c = UnsafeCell::new(45u32);
+        // }
```

Comment:
We could have this as a failing test, but it'd also be fine to not implement this in the first iteration. In fact I think that would be better -- let's take baby steps.